src/string/CreditCard.ts

```diff
@@ -0,0 +1,168 @@
+/**
+ * TODO: Add module comment
+ *
+ * @since 0.0.3
+ */
+import { Kind, Kind2, URIS, URIS2, HKT2 } from 'fp-ts/HKT'
+import * as D from 'io-ts/Decoder'
+import * as Enc from 'io-ts/Encoder'
+import * as Eq_ from 'fp-ts/Eq'
+import * as G from 'io-ts/Guard'
+import * as Str from 'fp-ts/string'
+import * as TD from 'io-ts/TaskDecoder'
+import * as t from 'io-ts/Type'
+import { flow, pipe } from 'fp-ts/function'
+import * as fc from 'fast-check'
+import * as RA from 'fp-ts/ReadonlyArray'
+
+import * as Arb from '../internal/ArbitraryBase'
+import { MonoidSum } from 'fp-ts/lib/number'
+import { digits } from '../internal/util'
+
+/**
+ * @since 0.0.3
+ * @category Internal
+ */
+interface CreditCardBrand {
+  readonly CreditCard: unique symbol
+}
+
+/**
+ * TODO: Add module comment
+ *
+ * @since 0.0.3
+ * @category Model
+ */
+export type CreditCard = string & CreditCardBrand
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams<S> = HKT2<S, string, CreditCard>
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams1<S extends URIS> = Kind<S, CreditCard>
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams2<S extends URIS2> = Kind2<S, string, CreditCard>
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams2C<S extends URIS2> = Kind2<S, unknown, CreditCard>
+
+const cc =
+  /(^4[0-9]{12}(?:[0-9]{3})?$)|(^(?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12}$)|(3[47][0-9]{13})|(^3(?:0[0-5]|[68][0-9])[0-9]{11}$)|(^6(?:011|5[0-9]{2})[0-9]{12}$)|(^(?:2131|1800|35\d{3})\d{11}$)/
+
+/** This calculates the Luhn checksum for a given string of digits. */
+const luhn: (cc: string) => number = flow(
```

Comment:
Annotation here would be nice as well, but I wonder if this algo might be used by other modules at some point? I wasn't familiar with it but [wiki says it has several uses](https://en.wikipedia.org/wiki/Luhn_algorithm).

Totally your call, if it makes sense we might drop this into `internal/algorithms` and add a couple tests. (Of course, this might be overkill).